main/parse.c

```diff
@@ -54,6 +55,8 @@ typedef struct {
 	enum specType specType;
 }  parserCandidate;
 
+static ptrArray *parsersUsedInCurrentInput;
```

Comment:
About setting NULL, static variables are set to zero, aren't they?
This may be about style but in my experience marking static is enough.

> 

Hum, actually it's probably better if the array is created and destroyed explicitly, isn't it? Right now it would be found as a still reachable leak in Valgrind, and IIRC you tried to get rid of those in the parsers, so I guess the same applies to the main part, doesn't it?

I have made too many `still reachable` in main part. Implementing finalizer or them and calling the finalizer  at the end of main() is not small change.

As you wrote I added such finalizer to parsers. However, as far as I can remember, I worked on eliminating  UNreachable leak.